openssl/src/pkcs7.rs

```diff
@@ -281,11 +296,47 @@ impl Pkcs7Ref {
             Ok(stack)
         }
     }
+
+    // Return the type of a PKCS#7 structure as a NID
+    pub fn type_nid(&self) -> Option<Nid> {
```

Comment:
It seems like this should just return the type, and leave the NID conversion to the caller.